api/client/attach.go

```diff
@@ -30,25 +30,30 @@ func (cli *DockerCli) CmdAttach(args ...string) error {
 		return err
 	}
 
-	env := engine.Env{}
-	if err := env.Decode(stream); err != nil {
+	//TODO: Should we reconsider having a type in api/types with a refactored runconfig?
```

Comment:
Agreed - why not have the entire 'real' strut defined?  Even if we're only using a few of the fields.